### ISSUE TITLE
Hot reload route templates

### ISSUE THREAD

**Issue by lifeart** [lifeart/demo-ember-vite#55]
Extending https://github.com/lifeart/demo-ember-vite/pull/54

Some refs about old implementations: https://github.com/lifeart/ember-ast-hot-load/blob/master/addon/services/hot-loader.js#L148

But at the moment, `renderTemplate` is not available https://github.com/emberjs/ember.js/pull/19659/files#diff-d933badede0c63e738ba4ef11ac2b7f9b4aa04b40cee6b8d71f97d0ba9a2ce63.

It seems we need to create wrapper for registry to handle it (now)

We could hook into `register` method of application and wrap route templates.

```js
  register(
    key: `${string}:${string}`,
    value: object | any,
    options?: any
  ): void {
    if (key.startsWith('template:')) {
      // hot-reload hook
      console.log('hot-reload', key, value);
    }
    return super.register(key, value, options);
  }
```

But, it seems (in general), we need more metadata for props forwarding.

Hot wrapper may looks like:
```hbs
<Hot @component={{template}} as |Template|>
    <Template @model={{@model}} />
</Hot>
```
In short, we need kinda props forwarding. And we have to collect it from original template. (`@model`, `this....`), but, `this` is not rally forward-able.

to reload route we could try to use `refresh` key
https://github.com/mixonic/ember.js/blob/8547049f81ec0f0928bc5786e9687f86f0f54f1a/packages/%40ember/-internals/routing/lib/system/route.ts#L1034
